src/python/foglamp/core/server.py

```diff
@@ -30,6 +43,197 @@ class Server:
+    def _run_management_api(cls):
+        web.run_app(cls._make_core(), host='0.0.0.0', port=8081)
+
+    @staticmethod
+    def get_management_pid():
+        """Returns FogLAMP's process id or None if FogLAMP is not running"""
+
+        try:
+            with open(_MANAGEMENT_PID_PATH, 'r') as pid_file:
+                pid = int(pid_file.read().strip())
+        except (IOError, ValueError):
+            return None
+
+        # Delete the pid file if the process isn't alive
+        # there is an unavoidable race condition here if another
+        # process is stopping or starting the daemon
+        try:
+            os.kill(pid, 0)
+        except OSError:
+            os.remove(_MANAGEMENT_PID_PATH)
+            pid = None
+
+        return pid
+
+    @classmethod
+    def stop_management(cls, pid=None):
+        """Stops Storage if it is running
+
+        Args:
+            pid: Optional process id to stop. If not specified, the pidfile is read.
+
+        Raises TimeoutError:
+            Unable to stop Storage. Wait and try again.
+        """
+        if not pid:
+            pid = cls.get_management_pid()
+
+        if not pid:
+            print("Management API is not running")
+            return
+
+        stopped = False
+
+        try:
+            for _ in range(_MAX_STOP_RETRY):
+                os.kill(pid, signal.SIGTERM)
+
+                for _ in range(_WAIT_STOP_SECONDS):  # Ignore the warning
+                    os.kill(pid, 0)
+                    time.sleep(1)
+                    os.remove(_MANAGEMENT_PID_PATH)
+        except OSError:
+            stopped = True
+
+        if not stopped:
+            raise TimeoutError("Unable to stop Management API")
+
+        print("Management API stopped")
+
+
+    """ Storage Services """
+    @staticmethod
+    def get_storage_pid():
+        """Returns Storage's process id or None if Storage is not running"""
+
+        try:
+            with open(_STORAGE_PID_PATH, 'r') as pid_file:
+                pid = int(pid_file.read().strip())
+        except (IOError, ValueError):
+            return None
+
+        # Delete the pid file if the process isn't alive
+        # there is an unavoidable race condition here if another
+        # process is stopping or starting the daemon
+        try:
+            os.kill(pid, 0)
+        except OSError:
+            os.remove(_STORAGE_PID_PATH)
+            pid = None
+
+        return pid
+
+    @classmethod
+    def start_storage(cls):
+        """Starts Storage"""
+
+        cls._safe_make_dirs(os.path.dirname(_STORAGE_PID_PATH))
+
+        pid = cls.get_storage_pid()
+
+        if pid:
+            print("Storage is already running in PID {}".format(pid))
+        else:
+            from subprocess import Popen, run
+            # TODO: Change below to use real Storage Service
+            p = Popen('foglamp/core/temp_test_storage.py')
+            # p.poll()
+
+            # Create storage pid in ~/var/run/storage.pid
```

Comment:
Please discuss with @MarkRiddoch regarding the use and purpose of pid files and the new directories ~/var/log and ~/var/run dirs. Verify that this is an acceptable approach to process management.